File: src/FlashcardParser/SingleLineParser/Alias.ts
import { LineDescriptor, IStudySet } from "../Types/Types";
import { PageRef } from "@/Commands/All/PageRef";

export function parseAcronymLine(
    lineDescriptor: LineDescriptor,
    studySet: IStudySet
): boolean {
    const { trimmedLine } = lineDescriptor;

    const parts = trimmedLine.split(PageRef.symbol);
    if (parts.length != 2) {
        console.log(`[parser] Alias is not in 2-part format`);
        return false;
    }
    const alias = parts[0];
    const value = parts[1];
    const aliasRecord = {} as Record<string, string>;
    aliasRecord[alias] = value;
    studySet.aliases.push(aliasRecord);

    return true;
}

File: service/flashcard_parser/single_line_parser/alias.py
from ..types import LineDescriptor, StudySet
from ..commands.page_ref import PageRef


def parse_acronym_line(line_descriptor: LineDescriptor, study_set: StudySet) -> bool:
    trimmed = line_descriptor.trimmed_line
    parts = trimmed.split(PageRef.symbol)
    if len(parts) != 2:
        print("[parser] Alias is not in 2-part format")
        return False
    alias, value = parts[0], parts[1]
    alias_record = {alias: value}
    study_set.aliases.append(alias_record)
    return True
